tests/generation/test_utils.py

```diff
@@ -136,6 +137,34 @@ def prepare_config_and_inputs_for_generate(self, batch_size=2):
 
         return config, filtered_inputs_dict
 
+    def _check_similar_generate_outputs(self, output_1, output_2, atol=1e-5, rtol=1e-5):
+        """
+        Checks whether a pair of generate outputs are similar. Two `generate` call outputs are considered similar in
+        the following siturations:
+        1. The sequences are the same
+        2. The sequences are different, but the scores up until the first mismatch are nearly identical
+        """
```

Comment:
Many `generate` tests wanted to check equivalence of something, and were incorrectly tagged as flaky. This function contains the correct equivalence check -- as a result, several `@is_flaky` were removed :D

(I honestly don't know why I haven't implemented this before, instead of adding `@is_flaky` 🙃 )